docs/en/observability/monitor-java-app.asciidoc

```diff
@@ -0,0 +1,2002 @@
+we didn't need to think about parsing our logs?
+
+Writing out logs as plain text works and is easy to read for humans.
+However, first writing them out as plain text, parsing them using
+the `dissect` processors, and then creating a JSON again sounds tedious and burns
+unneeded CPU cycles.
+
+While log4j2 has a
+https://logging.apache.org/log4j/2.x/manual/layouts.html#JSONLayout[JSONLayout],
+you can go further and use a Library called
+https://github.com/elastic/ecs-logging-java[ecs-logging-java]. The advantage of
+ECS logging is that it uses the https://www.elastic.co/guide/en/ecs/current/index.html[Elastic Common
+Schema]. ECS defines a standard set of fields used when storing event data in
+{es}, such as logs and metrics.
+
+. Instead of writing our logging standard, use an existing one. Let’s add
+the logging dependency to our Javalin application.
++
+[source,gradle]
+----
+dependencies {
+  implementation 'io.javalin:javalin:3.9.1'
+  implementation 'org.apache.logging.log4j:log4j-slf4j18-impl:2.13.3'
+  implementation 'co.elastic.logging:log4j2-ecs-layout:0.4.0'
+
+  testImplementation 'org.mockito:mockito-core:3.3.3'
+  testImplementation 'org.assertj:assertj-core:3.16.1'
+  testImplementation 'org.junit.jupiter:junit-jupiter-api:5.6.2'
+  testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.6.2'
+}
+----
++
+The log4j2-ecs-layout ships with a custom `<EcsLayout>` which can be used
+in the logging setup for the rolling file appender
++
+[source,xml]
+----
+<RollingFile name="JavalinAppLogRolling" fileName="/tmp/javalin/app.log" filePattern="/tmp/javalin/%d{yyyy-MM-dd}-%i.log.gz">
+  <EcsLayout serviceName="my-javalin-app"/>
+  <Policies>
+    <TimeBasedTriggeringPolicy />
+    <SizeBasedTriggeringPolicy size="50 MB"/>
+  </Policies>
+  <DefaultRolloverStrategy max="20"/>
+</RollingFile>
+----
++
+When you restart your app, you will see pure JSON written to your
+log file. When you trigger an exception, you will see, that the
+stack trace is already within your single document. This means the
+{filebeat} configuration can become stateless and even more lightweight.
+Also, the ingest pipeline on the {es} side can be deleted
+again.
++
+. You can configure a few
+https://github.com/elastic/ecs-logging-java/tree/master/log4j2-ecs-layout[more
+parameters] for the `EcsLayout`, but the defaults have been selected wisely. Let’s
+fix the {filebeat} configuration:
++
+[source,yaml]
+----
+filebeat.inputs:
+- type: log
+  enabled: true
+  paths:
+    - /tmp/javalin/*.log
+  json.keys_under_root: true
+
+name: javalin-app-shipper
+
+cloud.id: ${CLOUD_ID}
+output.elasticsearch:
+  api_key: ${ES_API_KEY}
+
+# ================================= Processors =================================
+processors:
+  - add_host_metadata: ~
+----
++
+As you can see, just by writing out logs as JSON, our whole logging
+setup got a ton easier, so whenever possible, try to directly
+write your logs as JSON.
```

Comment:
Upon rebuilding and rerunning, I get the following error:

```
Brandons-MacBook-Pro-4 elastic-obs-java-app $ java -jar build/libs/javalin-app-all.jar
ERROR StatusLogger Unrecognized format specifier [d]
ERROR StatusLogger Unrecognized conversion specifier [d] starting at position 16 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [thread]
ERROR StatusLogger Unrecognized conversion specifier [thread] starting at position 25 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [level]
ERROR StatusLogger Unrecognized conversion specifier [level] starting at position 35 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [logger]
ERROR StatusLogger Unrecognized conversion specifier [logger] starting at position 47 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [msg]
ERROR StatusLogger Unrecognized conversion specifier [msg] starting at position 54 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [n]
ERROR StatusLogger Unrecognized conversion specifier [n] starting at position 56 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [d]
ERROR StatusLogger Unrecognized conversion specifier [d] starting at position 16 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [thread]
ERROR StatusLogger Unrecognized conversion specifier [thread] starting at position 25 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [level]
ERROR StatusLogger Unrecognized conversion specifier [level] starting at position 35 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [logger]
ERROR StatusLogger Unrecognized conversion specifier [logger] starting at position 47 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [msg]
ERROR StatusLogger Unrecognized conversion specifier [msg] starting at position 54 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [n]
ERROR StatusLogger Unrecognized conversion specifier [n] starting at position 56 in conversion pattern.
```

I was able to fix this by adding the following to `build.gradle`:
```
shadowJar {
  transform(com.github.jengelman.gradle.plugins.shadow.transformers.Log4j2PluginsCacheFileTransformer)
}
```

We should add this instruction to the tutorial above.